pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -16,36 +16,82 @@ package cloudrun
 
 import (
 	"context"
+	"sync"
 
 	"go.uber.org/zap"
 
+	provider "github.com/pipe-cd/pipecd/pkg/app/piped/cloudprovider/cloudrun"
 	"github.com/pipe-cd/pipecd/pkg/config"
-	"github.com/pipe-cd/pipecd/pkg/model"
 )
 
-type applicationLister interface {
-	List() []*model.Application
+type store struct {
+	config        *config.CloudProviderCloudRunConfig
+	cloudProvider string
+	apps          map[string]provider.ServiceManifest
+	mu            sync.RWMutex
+	logger        *zap.Logger
+	client        provider.Client
 }
 
-type Store struct {
-	logger *zap.Logger
-}
-
-type Getter interface {
+func (s *store) addApp(sm provider.ServiceManifest) {
+	appID := sm.Labels()[provider.LabelApplication]
+	if appID == "" {
+		return
+	}
+	s.mu.Lock()
+	s.apps[appID] = sm
+	s.mu.Unlock()
 }
 
-func NewStore(cfg *config.CloudProviderCloudRunConfig, cloudProvider string, appLister applicationLister, logger *zap.Logger) *Store {
-	logger = logger.Named("cloudrun").
-		With(zap.String("cloud-provider", cloudProvider))
+func (s *store) run(ctx context.Context) error {
+	client, err := provider.DefaultRegistry().Client(ctx, s.cloudProvider, s.config, s.logger)
```

Comment:
How about initializing this client while the store is created and keeping it to use for the whole time instead of creating one each run like this?

Reply:
I thought [the same way](https://cyberagent.slack.com/archives/C02GANPH7S7/p1642750428156000) before, but when I looked at the implementation of `Client()`, it would use the cache if there was one hence I did like this.
However, I think it is more easier to understand that also initializing this client when the store is initialized.

Reply:
Therefore, , I will fix this after fixing it that ctx is passed in another PR.